### PR TITLE
Add an EntryCompletion example

### PR DESCRIPTION
**Pull request by apiraino**
I spend some fun trying to figure out how the autocompletion of a field work. I thought to build a basic example. Make sense?

Can you help me improve this example?

Thanks!

### CODE REVIEW

File: src/bin/entry_completion.rs [hunk 1]
```diff
@@ -0,0 +1,110 @@
+extern crate glib;
```
Comment (GuillaumeGomez) [inline src/bin/entry_completion.rs]: Please add a small explanation of what's going on in this example at the top of the file. :)

**Comment by GuillaumeGomez** [comment]
Thanks for the PR! I changed the base branch to pending (all new PRs should target this branch in this repository). You'll need to rebase. Also, the code looked good to me. Maybe @EPashkin and @sdroege have things to say?

File: src/bin/entry_completion.rs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+    let data: [Data; 4] = [
+        Data {description: "France".to_string()},
+        Data {description: "Italy".to_string()},
+        Data {description: "Siberia".to_string()},
+        Data {description: "Switzerland".to_string()},
+    ];
+    let store = gtk::ListStore::new(&col_types);
+    let col_indices: [u32; 1] = [0];
+    for (_, d) in data.iter().enumerate() {
```
Comment (sdroege) [inline src/bin/entry_completion.rs]: Why use `enumerate()` if you ignore the index it generates anyway?

File: src/bin/entry_completion.rs [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+fn create_list_model() -> gtk::ListStore {
+    let col_types: [gtk::Type; 1] = [
+        gtk::Type::String,
+    ];
+
+    let data: [Data; 4] = [
+        Data {description: "France".to_string()},
+        Data {description: "Italy".to_string()},
+        Data {description: "Siberia".to_string()},
```
Comment (sdroege) [inline src/bin/entry_completion.rs]: Last time I checked, Siberia was not a country ;)

**Comment by sdroege** [comment]
Looks good to me otherwise, thanks!

File: src/bin/entry_completion.rs (hunk 2 above)
Comment (apiraino) [reply src/bin/entry_completion.rs]: copy-pasting like there's not tomorrow :100: 

fixed!

**Comment by apiraino** [comment]
ok, we should be done (if I didn't mess up things rebasing). I've noticed there's an example failing to compile (`transparent_main_window.rs`) on the `pending` branch but I can't fix it ATM - and probably not relevant to this PR

**Comment by GuillaumeGomez** [comment]
Thanks!

**Comment by sdroege** [comment]
>  I've noticed there's an example failing to compile (`transparent_main_window.rs`) on the `pending` branch but I can't fix it ATM

`cargo update`, you need a newer version of the gtk crate :)

**Pull request merged**